frb_codegen/src/commands.rs

```diff
@@ -274,3 +280,91 @@ pub fn build_runner(dart_root: &str) -> Result {
     }
     Ok(())
 }
+
+fn guess_context(dart_root: &str) -> anyhow::Result<Context> {
+    debug!("Guessing context the runner is run into");
+    let pubspec = PathBuf::from(dart_root).join("pubspec.lock");
+    if !pubspec.exists() {
+        return Err(anyhow::Error::msg(format!(
+            "missing pubspec.lock in {}",
+            dart_root
+        )));
+    }
+    let pubspec = std::fs::read_to_string(pubspec)
+        .map_err(|_| anyhow::Error::msg(format!("unable to read pubspec.lock in {}", dart_root)))?;
+    let pubspec: Pubspec = serde_yaml::from_str(&pubspec).map_err(|_| {
+        anyhow::Error::msg(format!("unable to parse pubspec.lock in {}", dart_root))
+    })?;
+    if pubspec.packages.contains_key("flutter") {
+        return Ok(Context::Flutter);
+    }
+    Ok(Context::Dart)
+}
+
+#[derive(Debug, PartialEq)]
+enum Context {
+    Dart,
+    Flutter,
+}
+
+impl Context {
+    fn as_run_command(&self) -> &'static str {
+        match self {
+            Context::Dart => "dart",
+            Context::Flutter => "flutter pub",
+        }
+    }
+}
+
+#[derive(Debug, Deserialize)]
+struct Pubspec {
```

Comment:
uber-nit: `PubspecLock` (since o/w people may think it is pubspec yaml not pubspec lock)

Reply:
Ah yes this makes sense, I'm on it.

Reply:
Done in e241da7321605672cf0d70ab178dfbd49bdbfb51